ui/src/routes/shared.$sharedToken.lazy.tsx

```diff
@@ -92,15 +95,20 @@ const EditorComponent = ({ accessToken }: { accessToken?: string }) => {
 
   const { sharedToken } = useParams({ strict: false });
 
-  const { sharedProject } = useGetSharedProject(sharedToken);
+  const { sharedProject, isError } = useGetSharedProject(sharedToken);
 
   const { yWorkflows, yDocState, isSynced, undoTrackerActionWrapper } =
     useYjsSetup({
       projectId: sharedProject?.id,
       workflowId: DEFAULT_ENTRY_GRAPH_ID,
     });
 
-  return !yWorkflows || !isSynced || !undoTrackerActionWrapper ? (
+  return isError ? (
+    <NotFound />
```

Comment:
NotFound doesn't really apply to isError in useGetSharedProject, does it? NotFound should be only used if a route is wrong as a fallback. Not for errors.